GameServerLib/Logic/Config.cs

```diff
@@ -13,22 +14,33 @@ public class Config
         public MapSpawns MapSpawns { get; private set; }
         public ContentManager ContentManager { get; private set; }
         public const string VERSION = "Version 4.20.0.315 [PUBLIC]";
+        private Logger _logger;
 
         public bool CooldownsEnabled { get; private set; }
         public bool ManaCostsEnabled { get; private set; }
         public bool ChatCheatsEnabled { get; private set; }
         public bool MinionSpawnsEnabled { get; private set; }
 
-        public Config(string path)
+        public Config(Logger logger, string path, string inputJson = "")
```

Comment:
Okay, two things:
* Ninject should take care of passing Logger to here, so we don't actually need to pass it manually.
* We want a way to create a config with just the json data, no file paths and/or data

```CS
private Config(Logger logger)
{
    _logger = logger;
}

public static Config LoadFromJson(string json)
{
    result = new Config(Program.ResolveDependency<Logger>());
    result.LoadConfig(json);
    return result;
}

private void LoadConfig(string json)
{
...
```
After this we can create the config object already in `GameServerApp` just by doing
```CS
config = Config.LoadConfig(json):
```

The Ninject part there is `Program.ResolveDependency<Logger>()`, which is marked deprecated (for a reason). It should really autoresolve the dependency, but since the constructor is private, it'd need some more work that can be put off to a later date. If you want to read more about ninject, https://github.com/ninject/Ninject/wiki/Dependency-Injection-With-Ninject is a good place to start.